src/v/transform/io.h

```diff
@@ -57,7 +57,7 @@ class source {
      * The offset of a record the log for a given timestamp - if the log is
      * empty then `kafka::offset::min()` is returned.
      */
-    virtual ss::future<kafka::offset>
+    virtual ss::future<std::optional<kafka::offset>>
```

Comment:
Can we update the docs (the comment about min is obsolete for example). What does std::nullopt mean? Can we confirm with the storage team?

Reply:
>update docs

ya, looks like I smashed the changed while shuffling commits

>what does std::nullopt mean

local case:

https://github.com/redpanda-data/redpanda/blob/d692443549b2b96593947064f7811ff08989b90d/src/v/storage/disk_log_impl.cc#L2450-L2470

I read that as one of:

a) log empty 
b) no batches in timequery request offset range
c) max_timestamp in read batches < timequery request timestamp

which tracks I think? "empty query result".

Reply:
(b) doesn't really apply here by design, and `latest` seems like a sensible fallback for (a) or (c)

Reply:
LGTM